File: PLACEMENT_PREPARATION/ZOHO-2025/Gopinath/Day1/one.py
from typing import List
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        n=len(matrix)
        for i in range(n):
            matrix[i]=matrix[i][::-1]
        for i in range(n-1):
            for j in range(n-i-1):
                matrix[i][j],matrix[-j-1][-i-1]=matrix[-j-1][-i-1],matrix[i][j]

File: PLACEMENT_PREPARATION/ZOHO-2025/keerthi/Day1/One.java
// Implement an algorithm to rotate an NxN matrix by 90 degrees. 

class Solution {
    public void rotate(int[][] matrix) {

        int[][] res = new int[matrix.length][matrix.length];

        
        int  right = matrix.length-1;

        for(int i = 0;i<matrix.length;i++){
            
            for(int j=0;j<matrix[i].length;j++){
        
              res[j][right] = matrix[i][j];
          
            }
            right--;
            
        }

        for (int i = 0; i < res.length; i++) {
            for (int j = 0; j < res[i].length; j++) {
                
                matrix[i][j] = res[i][j];
            }
          
        }
    }


//keerthi
